src/NuProj.Tests/GenerateNuSpecTests.cs

```diff
@@ -197,5 +198,81 @@ public void Task_GenerateNuSpec_UseInputFileName()
                     ManifestFileComparer.Instance);
             }
         }
+
+        [Theory]
+        [InlineData("Task_GenerateNuSpec_NoRequiredParameters", @"NuGetPackage\NuGetPackage.nuproj")]
+        public async System.Threading.Tasks.Task Task_GenerateNuSpec_NoRequiredParameters(string scenarioName, string projectToBuild)
+        {
+            var solutionDir = Assets.GetScenarioDirectory(scenarioName);
+            var projectPath = Path.Combine(solutionDir, projectToBuild);
+
+            var result = await MSBuild.ExecuteAsync(projectPath, "GenerateNuSpec");
+            result.AssertError("The 'Id' property cannot be empty or undefined.");
+        }
+
+        [Theory]
+        [InlineData("Task_GenerateNuSpec_NoRequiredParameters", @"NuGetPackage\NuGetPackage.nuproj")]
+        public async System.Threading.Tasks.Task Task_GenerateNuSpec_NoRequiredParameters_IdOnly(string scenarioName, string projectToBuild)
+        {
+            var solutionDir = Assets.GetScenarioDirectory(scenarioName);
+            var projectPath = Path.Combine(solutionDir, projectToBuild);
+
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                {"Id", "NoRequiredParameters"}
+            };
+
+            var result = await MSBuild.ExecuteAsync(projectPath, "GenerateNuSpec", properties);
+            result.AssertError("The 'Version' property cannot be empty or undefined.");
```

Comment:
I'm getting `The "GenerateNuSpec" task was not given a value for the required parameter "Id".` here.